README.md

```diff
@@ -106,3 +107,13 @@ main = evalZ3 script >>= \mbSol ->
              Nothing  -> error "No solution found."
              Just sol -> putStr "Solution: " >> print sol
 ```
+
+## Garbage Collection
+
+This library automatically garbage collects all C objects created through its API.
+
+## Concurrency
+
+If you intend to use this library's API from multiple threads,
```

Comment:
Once you've added synchronization to any `Context`-manipulating function, haven't we made the library thread-safe?

Reply:
Oh yes, you are right of course!